S3 unit test build broken: gtl not declared

<em>Please make sure that this is a bug. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): No
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 16.04
- TensorFlow installed from (source or binary): source
- TensorFlow version (use command below): master @3d79d19aa27abd1e236761a6071acec858acb6a0
- Python version: 3.6
- Bazel version (if compiling from source): 1.1.0
- GCC/Compiler version (if compiling from source): gcc5
- CUDA/cuDNN version: NA
- GPU model and memory: NA

**Describe the current behavior**
Test fails to build
```
ERROR: /tensorflow/tensorflow/core/platform/s3/BUILD:96:1: C++ compilation of rule '//tensorflow/core/platform/s3:s3_file_system_test' failed (Exit 1)
In file included from ./tensorflow/core/platform/env.h:27:0,
                 from ./tensorflow/core/platform/s3/s3_file_system.h:20,
                 from tensorflow/core/platform/s3/s3_file_system_test.cc:16:
tensorflow/core/platform/s3/s3_file_system_test.cc: In member function 'tensorflow::Status tensorflow::{anonymous}::S3FileSystemTest::ReadAll(const string&, std::__cxx11::string*)':
tensorflow/core/platform/s3/s3_file_system_test.cc:59:45: error: 'gtl' has not been declared
         reader->Read(0, file_size, &result, gtl::string_as_array(content)));
                                             ^
./tensorflow/core/platform/errors.h:73:37: note: in definition of macro 'TF_RETURN_IF_ERROR'
     ::tensorflow::Status _status = (__VA_ARGS__);        \
                                     ^~~~~~~~~~~
In file included from external/com_google_googletest/googletest/include/gtest/gtest.h:388:0,
                 from ./tensorflow/core/platform/test.h:31,
                 from ./tensorflow/core/lib/core/status_test_util.h:20,
                 from tensorflow/core/platform/s3/s3_file_system_test.cc:18:
tensorflow/core/platform/s3/s3_file_system_test.cc: In member function 'virtual void tensorflow::{anonymous}::S3FileSystemTest_NewRandomAccessFile_Test::TestBody()':
tensorflow/core/platform/s3/s3_file_system_test.cc:83:48: error: 'gtl' has not been declared
       reader->Read(0, content.size(), &result, gtl::string_as_array(&got)));
                                                ^
tensorflow/core/platform/s3/s3_file_system_test.cc:82:3: note: in expansion of macro 'TF_EXPECT_OK'
   TF_EXPECT_OK(
   ^~~~~~~~~~~~
tensorflow/core/platform/s3/s3_file_system_test.cc:89:44: error: 'gtl' has not been declared
   TF_EXPECT_OK(reader->Read(2, 4, &result, gtl::string_as_array(&got)));
                                            ^
tensorflow/core/platform/s3/s3_file_system_test.cc:89:3: note: in expansion of macro 'TF_EXPECT_OK'
   TF_EXPECT_OK(reader->Read(2, 4, &result, gtl::string_as_array(&got)));
   ^~~~~~~~~~~~
Target //tensorflow/core/platform/s3:s3_file_system_test failed to build
Use --verbose_failures to see the command lines of failed build steps.
INFO: Elapsed time: 5.538s, Critical Path: 5.20s
INFO: 0 processes.
FAILED: Build did NOT complete successfully
```
**Describe the expected behavior**
Test builds and runs

**Code to reproduce the issue**
bazel test -c opt //tensorflow/core/platform/s3:s3_file_system_test --action_env=S3_TEST_TMPDIR=s3://BUCKET/s3testing-results/bazel/master/

**Other info / logs**
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached.

--- a/tensorflow/core/platform/s3/s3_file_system_test.cc
+++ b/tensorflow/core/platform/s3/s3_file_system_test.cc
@@ -55,7 +55,7 @@ class S3FileSystemTest : public ::testing::Test {
     content->resize(file_size);
     StringPiece result;
     TF_RETURN_IF_ERROR(
-        reader->Read(0, file_size, &result, gtl::string_as_array(content)));
+        reader->Read(0, file_size, &result, &(*content)[0]));
     if (file_size != result.size()) {
       return errors::DataLoss("expected ", file_size, " got ", result.size(),
                               " bytes");

--- a/tensorflow/core/platform/s3/s3_file_system_test.cc
+++ b/tensorflow/core/platform/s3/s3_file_system_test.cc
@@ -79,13 +79,13 @@ TEST_F(S3FileSystemTest, NewRandomAccessFile) {
   got.resize(content.size());
   StringPiece result;
   TF_EXPECT_OK(
-      reader->Read(0, content.size(), &result, gtl::string_as_array(&got)));
+      reader->Read(0, content.size(), &result, &got[0]));
   EXPECT_EQ(content.size(), result.size());
   EXPECT_EQ(content, result);
 
   got.clear();
   got.resize(4);
-  TF_EXPECT_OK(reader->Read(2, 4, &result, gtl::string_as_array(&got)));
+  TF_EXPECT_OK(reader->Read(2, 4, &result, &got[0]));
   EXPECT_EQ(4, result.size());
   EXPECT_EQ(content.substr(2, 4), result);
 }

